tests/mocha/icon_test.js

```diff
@@ -22,38 +22,75 @@ suite.skip('Icon', function () {
     sharedTestTeardown.call(this);
   });
 
-  suite('Hooks getting properly triggered by the block', function () {
-    class MockIcon {
-      initView() {}
+  class MockIcon {
+    getType() {
+      return 'mock icon';
+    }
+
+    initView() {}
+
+    applyColour() {}
 
-      applyColour() {}
+    updateEditable() {}
 
-      updateEditable() {}
+    updateCollapsed() {}
+  }
 
-      updateCollapsed() {}
+  class MockSerializableIcon extends MockIcon {
+    constructor() {
+      super();
+      this.state = '';
     }
 
-    function createHeadlessWorkspace() {
-      return new Blockly.Workspace();
+    getType() {
+      return 'serializable icon';
     }
 
-    function createWorkspaceSvg() {
-      const workspace = new Blockly.WorkspaceSvg(new Blockly.Options({}));
-      workspace.createDom();
-      return workspace;
+    getWeight() {
+      return 1;
     }
 
-    function createUninitializedBlock(workspace) {
-      return workspace.newBlock('empty_block');
+    saveState() {
+      return 'some state';
     }
 
-    function createInitializedBlock(workspace) {
-      const block = workspace.newBlock('empty_block');
-      block.initSvg();
-      block.render();
-      return block;
+    loadState(state) {
+      this.state = state;
     }
+  }
 
+  class MockNonSerializableIcon extends MockIcon {
+    getType() {
+      return 'non-serializable icon';
+    }
+  }
+
+  function createHeadlessWorkspace() {
+    return new Blockly.Workspace();
+  }
+
+  function createWorkspaceSvg() {
+    const workspace = new Blockly.WorkspaceSvg(new Blockly.Options({}));
+    workspace.createDom();
+    return workspace;
+  }
+
+  function createUninitializedBlock(workspace) {
+    return workspace.newBlock('empty_block');
+  }
+
+  function createInitializedBlock(workspace) {
+    const block = workspace.newBlock('empty_block');
+    block.initSvg();
+    block.render();
+    return block;
+  }
+
+  function createHeadlessBlock(workspace) {
+    return createUninitializedBlock(workspace);
+  }
+
+  suite.skip('Hooks getting properly triggered by the block', function () {
```

Comment:
Is this still supposed to be skipped?

Reply:
Yes, this still hasn't been implemented. It'll get fixed when I migrate over the first built-in icon (or right before).